service/controller/clusterapi/v17/resources/machinedeploymentstatus/resource.go

```diff
@@ -81,9 +83,17 @@ func (r *Resource) ensure(ctx context.Context, obj interface{}) error {
 			return nil
 		}
 
-		l, err := cc.Client.TenantCluster.K8s.CoreV1().Nodes().List(metav1.ListOptions{})
+		o := metav1.ListOptions{
+			// This label selector excludes the master nodes from node list.
+			//
+			// Constructing this LabelSelector is not currently possibly with
+			// types & functions in k8s. Therefore it's hardcoded here.
+			LabelSelector: fmt.Sprintf("!%s", label.MasterNodeRole),
```

Comment:
Exclude masters from node list.

Reply:
What is somehow not cool with this is the external dependency on the label definition. Tomorrow we will have forgotten how this works and some obscure change with the next k8s release will fuck around with this and hefty debugging starts. Any idea how to not let this slip through @giantswarm/team-ludacris?